.github/scripts/transform_dbt_test_results.py

```diff
@@ -0,0 +1,1082 @@
+            # Athena SHOW COLUMNS doesn't allow double quoted tablenames
+            relation_name_unquoted = test_results_relation_name.replace(
+                '"', "`"
+            )
+            cursor.execute(f"show columns in {relation_name_unquoted}")
+            # SHOW COLUMNS often returns field names with trailing whitespace
+            fieldnames = [row["field"].strip() for row in cursor]
+
+            test_results_query = f"select * from {test_results_relation_name}"
+            if (
+                PARID_FIELD in fieldnames
+                and TAXYR_FIELD in fieldnames
+                and TOWNSHIP_FIELD not in fieldnames
+            ):
+                # If parid and taxyr are present, try to retrieve the township
+                # code for every row. It's most efficient to do this via a
+                # join in the query rather than in a Python lookup since
+                # legdat has 43m rows
+                test_results_query = f"""
+                    select test_results.*, leg.user1 as {TOWNSHIP_FIELD}
+                    from {test_results_relation_name} as test_results
+                    left join iasworld.legdat as leg
+                        on leg.{PARID_FIELD} = test_results.{PARID_FIELD}
+                        and leg.{TAXYR_FIELD} = test_results.{TAXYR_FIELD}
+                """
+
+            cursor.execute(test_results_query)
+            query_results = cursor.fetchall()
+            if len(query_results) == 0:
+                raise ValueError(
+                    f"Test {test_name} has status 'fail' but no failing rows "
+                    "in Athena"
+                )
+
+            # Add custom fields to query results that we don't expect to be
+            # included in the response
+            failing_rows = [
+                {
+                    TEST_NAME_FIELD: test_name,
+                    DESCRIPTION_FIELD: test_description,
+                    DOCS_URL_FIELD: test_docs_url,
+                    SOURCE_TABLE_FIELD: tablename,
+                    **row,
+                }
+                for row in query_results
+            ]
+            test_result = TestResult(
+                status=Status.FAIL,
+                failing_rows=failing_rows,
+                **base_result_kwargs,
+            )
+
+            if not tests_by_category.get(category):
+                tests_by_category[category] = TestCategory(category=category)
+            tests_by_category[category].test_results.append(test_result)
+
+        else:
+            raise ValueError(
+                f"Got unrecognized status '{status}' for node {unique_id} "
+                "in dbt run results"
+            )
+
+    # Now that we've accumulated all of the test results and they are grouped
+    # into categories, we no longer need the category key in the dict, so
+    # transform the output into a list
+    return list(tests_by_category.values())
+
+
+def get_category_from_node(node: typing.Dict) -> str:
```

Comment:
`get_category_from_node` and `get_tablename_from_node` are untouched since the last version of this script.